app/src/test/java/com/codesoom/assignment/service/CatToyServiceTest.java

```diff
@@ -0,0 +1,202 @@
+                        .forEach(index -> {
+                            CatToy catToy = result.get(index - 1);
+                            assertThat(catToy.getName()).isEqualTo(FIXTURE_NAME + index);
+                            assertThat(catToy.getMaker()).isEqualTo(FIXTURE_MAKER + index);
+                            assertThat(catToy.getPrice()).isEqualTo(FIXTURE_PRICE + index);
+                            assertThat(catToy.getImageURL()).isEqualTo(FIXTURE_IMAGE_URL + index);
+                        });
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("findById 메소드는")
+    class Describe_findById {
+        @Nested
+        @DisplayName("저장되지 않은 장난감의 Id로 요청했을 떄")
+        class Context_withIdOfNotSavedToy {
+            @BeforeEach
+            void prepare() {
+                repository.deleteAll();
+            }
+
+            @Test
+            @DisplayName("null 값을 반환한다")
+            void it_returnsFoundToy() {
+                Optional<CatToy> result = service.findById(1L);
+
+                assertThat(result).isEmpty();
+            }
+        }
+
+        @Nested
+        @DisplayName("저장되어 있는 장난감의 Id로 요청했을 떄")
+        class Context_withIdOfSavedToy {
+            private final CatToy savedCatToy = new CatToy(FIXTURE_NAME, FIXTURE_MAKER, FIXTURE_PRICE, FIXTURE_IMAGE_URL);
+
+            @BeforeEach
+            void prepare() {
+                repository.save(savedCatToy);
+            }
+
+            @Test
+            @DisplayName("조회된 장난감을 반환한다")
+            void it_returnsFoundToy() {
+                final Long toyId = 1L;
+                Optional<CatToy> result = service.findById(toyId);
+
+                assertThat(result).isPresent();
+                assertThat(result.get().getId()).isEqualTo(toyId);
+                assertThat(result.get().getName()).isEqualTo(FIXTURE_NAME);
+                assertThat(result.get().getMaker()).isEqualTo(FIXTURE_MAKER);
+                assertThat(result.get().getPrice()).isEqualTo(FIXTURE_PRICE);
+                assertThat(result.get().getImageURL()).isEqualTo(FIXTURE_IMAGE_URL);
+            }
+        }
+    }
+
+    @Nested
+    @DisplayName("save 메소드는")
+    class Describe_save {
+        @Nested
+        @DisplayName("필수 값들을 가지고 있는 장난감으로 요청했을 때")
+        class Context_withToyHasRequiredFields {
+            final CatToy toyWithRequiredFields = new CatToy(FIXTURE_NAME, FIXTURE_MAKER, FIXTURE_PRICE, FIXTURE_IMAGE_URL);
+
+            @Test
+            @DisplayName("저장된 장난감을 반환한다")
+            void it_returnsSavedToy() {
+                CatToy result = service.save(toyWithRequiredFields);
+
+                assertThat(result.getId()).isEqualTo(1L);
+                assertThat(result.getName()).isEqualTo(FIXTURE_NAME);
+                assertThat(result.getMaker()).isEqualTo(FIXTURE_MAKER);
+                assertThat(result.getPrice()).isEqualTo(FIXTURE_PRICE);
+                assertThat(result.getImageURL()).isEqualTo(FIXTURE_IMAGE_URL);
```

Comment:
```suggestion
                assertThat(result).hasValue(savedCatToy);
```
해당 코드가 계속 겹쳐서 보이는군요! 
위 코드가 성공하도록 변경해도 되고 만드신 코드를 함수로 분리해도 좋을 것 같아요!

Reply:
`assertThat(result)`의 return type이 ObjectAssert 인데 
혹시 자바에서 ObjectAssert의 기능을 확장하는 방법이 있을까요?

swift에서는 아래와 같은 방식이 가능한데, java에도 이런 방법이 있어서 제안주신건지 궁금합니다.

```swift
class ObjectAssert { }

extension ObjectAssert where T: CatToy {
    func hasValue(_ other: T) {
        
    }
}
```

혹시 방법이 있다면 키워드 주시면 확인해볼게요!

Reply:
아 ObjectAssert를 상속 받는 방법도 있겠군요

```java
public class CatToyAssert extends ObjectAssert<CatToy> {
    public CatToyAssert(CatToy o) {
        super(o);
    }

    public void hasValues(CatToy expected) {

    }
}
```

Reply:
여러가지 방법이 있을 수 있지만 자바에서는 `equals` 를 구현하는 것으로 동일한지 여부를 판별합니다!

또한 `equals` 를 구현하면 자연스럽게 `hashcode` 하는 메소드도 구현해야 해요!

왜 구현해야하는지 찾아보면 좋은 경험이 될 거에요!